lib/tapioca/gem/listeners/methods.rb

```diff
@@ -66,7 +66,7 @@ def compile_directly_owned_methods(tree, module_name, mod, for_visibility = [:pu
         end
         def compile_method(tree, symbol_name, constant, method, visibility = RBI::Public.new)
           return unless method
-          return unless method.owner == constant
+          return unless method_defined_by_constant?(method, constant)
```

Comment:
I know that I had named this method but looking at it again, it feels like `method_owned_by_constant?` would be a more appropriate name, since we are doing ownerships checks.